presto-main/src/main/java/io/prestosql/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -165,6 +178,78 @@ private void analyze(Expression expression)
         }
     }
 
+    private static Expression normalize(Expression expression)
+    {
+        return ExpressionTreeRewriter.rewriteWith(new ExpressionNormalizer(), expression);
+    }
+
+    private static class ExpressionNormalizer
+            extends ExpressionRewriter<Void>
+    {
+        @Override
+        public Expression rewriteCast(Cast node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            return new Cast(
```

Comment:
Is cast really the only expression with this issue?

Reply:
perhaps not, but we need to start somewhere.